regexp.go

```diff
@@ -312,17 +312,13 @@ func (v *routeRegexpGroup) setMatch(req *http.Request, m *RouteMatch, r *Route)
 }
 
 // getHost tries its best to return the request host.
+// According to section 14.23 of RFC 2616 the Host header
+// can include the port number if the default value of 80 is not used.
```

Comment:
Can you point me to the relevant section of the RFC that covers this requirement for the Host header? It is mentioned in the context of a URL, but seems more ambiguous in the context of a Host header. It is not immediately clear in 7230 either: https://tools.ietf.org/html/rfc7230#section-5.4

> 3.2.2 http URL
> The "http" scheme is used to locate network resources via the HTTP
   protocol. This section defines the scheme-specific syntax and
   semantics for http URLs.
>  http_URL = "http:" "//" host [ ":" port ] [ abs_path [ "?" query ]]
> If the port is empty or not given, port 80 is assumed. 

I'm cautious about making changes here as it may change the matching behaviour (read: breaking change) of existing users.